tests/scanners/test.javascript.js

```diff
@@ -99,54 +98,45 @@ describe('JavaScript Scanner', () => {
     ]);
   });
 
-  it('should pass when async/await is used', () => {
+  it('should pass when async/await is used', async () => {
     const code = 'var foo = async a => a;';
     const jsScanner = new JavaScriptScanner(code, 'code.js');
 
-    return jsScanner.scan()
-      .then(({ linterMessages }) => {
-        expect(linterMessages.length).toEqual(0);
-      });
+    const { linterMessages } = await jsScanner.scan();
+    expect(linterMessages.length).toEqual(0);
   });
 
-  it('should support object spread syntax', () => {
+  it('should support object spread syntax', async () => {
     const code = oneLine`
       const config = {};
       const actual = {...config, foo: 'bar'};
     `;
 
     const jsScanner = new JavaScriptScanner(code, 'code.js');
 
-    return jsScanner.scan()
-      .then(({ linterMessages }) => {
-        expect(linterMessages.length).toEqual(0);
-      });
+    const { linterMessages } = await jsScanner.scan();
+    expect(linterMessages.length).toEqual(0);
   });
 
-  it('should create an error message when encountering a syntax error', () => {
-    let code = 'var m = "d;';
-    let jsScanner = new JavaScriptScanner(code, 'badcode.js');
-
-    return jsScanner.scan()
-      .then(({ linterMessages }) => {
-        expect(linterMessages[0].code).toEqual(messages.JS_SYNTAX_ERROR.code);
-        expect(linterMessages[0].type).toEqual(VALIDATION_ERROR);
-
-        // Test another error for good measure.
-        code = 'var aVarThatDoesnt != exist;';
-        jsScanner = new JavaScriptScanner(code, 'badcode.js');
-
-        return jsScanner.scan()
-          .then(({ linterMessages: moreValidationMessages }) => {
-            expect(moreValidationMessages[0].code).toEqual(
-              messages.JS_SYNTAX_ERROR.code);
-            expect(moreValidationMessages[0].type).toEqual(
-              VALIDATION_ERROR);
-          });
-      });
+  it('should create an error message when encountering a syntax error (option 1)', async () => {
+    const code = 'var m = "d;';
+    const jsScanner = new JavaScriptScanner(code, 'badcode.js');
+
+    const { linterMessages } = await jsScanner.scan();
+    expect(linterMessages[0].code).toEqual(messages.JS_SYNTAX_ERROR.code);
+    expect(linterMessages[0].type).toEqual(VALIDATION_ERROR);
+  });
+
+  it('should create an error message when encountering a syntax error (option 2)', async () => {
```

Comment:
@rpl it's just the only test in the whole suite with 2 test cases, so I thought it would be cleaner to split them into 2. But if you think it's ok, I'll move them into 1

Reply:
Yeah, I think that it makes sense to merge them again, the "(option 1)" and "(options 2)" suffixes in the test description seem also to suggest that they are actually two variants of the same scenario.